conan/api/subapi/cache.py

```diff
@@ -131,50 +130,48 @@ def save(self, package_list, tgz_path, no_source=False):
-                # Package only selected folders, not DOWNLOAD one
-                for f in (EXPORT_FOLDER, EXPORT_SRC_FOLDER, SRC_FOLDER):
-                    if f == SRC_FOLDER and no_source:
-                        continue
-                    path = os.path.join(cache_folder, recipe_folder, f)
-                    if os.path.exists(path):
-                        tgz.add(path, f"{recipe_folder}/{f}", recursive=True)
-                path = os.path.join(cache_folder, recipe_folder, DOWNLOAD_EXPORT_FOLDER, METADATA)
+        tar_files: dict[str,str] = {} # {path_in_tar: abs_path}
+
+        for ref, ref_bundle in package_list.refs().items():
+            ref_layout = cache.recipe_layout(ref)
+            recipe_folder = os.path.relpath(ref_layout.base_folder, cache_folder)
+            recipe_folder = recipe_folder.replace("\\", "/")  # make win paths portable
+            ref_bundle["recipe_folder"] = recipe_folder
+            out.info(f"Saving {ref}: {recipe_folder}")
+            # Package only selected folders, not DOWNLOAD one
+            for f in (EXPORT_FOLDER, EXPORT_SRC_FOLDER, SRC_FOLDER):
+                if f == SRC_FOLDER and no_source:
+                    continue
+                path = os.path.join(cache_folder, recipe_folder, f)
                 if os.path.exists(path):
-                    tgz.add(path, f"{recipe_folder}/{DOWNLOAD_EXPORT_FOLDER}/{METADATA}",
-                            recursive=True)
-
-                for pref, pref_bundle in package_list.prefs(ref, ref_bundle).items():
-                    pref_layout = cache.pkg_layout(pref)
-                    pkg_folder = pref_layout.package()
-                    folder = os.path.relpath(pkg_folder, cache_folder)
-                    folder = folder.replace("\\", "/")  # make win paths portable
-                    pref_bundle["package_folder"] = folder
-                    out.info(f"Saving {pref}: {folder}")
-                    tgz.add(os.path.join(cache_folder, folder), folder, recursive=True)
-                    if os.path.exists(pref_layout.metadata()):
-                        metadata_folder = os.path.relpath(pref_layout.metadata(), cache_folder)
-                        metadata_folder = metadata_folder.replace("\\", "/")  # make paths portable
-                        pref_bundle["metadata_folder"] = metadata_folder
-                        out.info(f"Saving {pref} metadata: {metadata_folder}")
-                        tgz.add(os.path.join(cache_folder, metadata_folder), metadata_folder,
-                                recursive=True)
-            serialized = json.dumps(package_list.serialize(), indent=2)
-            info = tarfile.TarInfo(name="pkglist.json")
-            data = serialized.encode('utf-8')
-            info.size = len(data)
-            tgz.addfile(tarinfo=info, fileobj=BytesIO(data))
-            tgz.close()
+                    tar_files.update({f"{recipe_folder}/{f}": path})
```

Comment:
```suggestion
                    tar_files[f"{recipe_folder}/{f}"] = path
```

Same for every other ``tar_files.update()``

Reply:
Yep, definitely I forgot to code in python after holidays 🐸